src/MassTransit.ActiveMqTransport/Configuration/Configurators/ConfigurationHostSettings.cs

```diff
@@ -14,92 +14,47 @@ namespace MassTransit.ActiveMqTransport.Configurators
 {
     using System;
     using Apache.NMS;
-    using Apache.NMS.ActiveMQ;
-    using Apache.NMS.ActiveMQ.Transport;
-    using Apache.NMS.ActiveMQ.Transport.Tcp;
-    using Apache.NMS.ActiveMQ.Util;
 
 
     public class ConfigurationHostSettings :
         ActiveMqHostSettings
     {
         readonly Lazy<Uri> _hostAddress;
         readonly Lazy<Uri> _brokerAddress;
+        readonly Lazy<IConnectionFactory> _connectionFactory;
 
         public ConfigurationHostSettings()
         {
             _hostAddress = new Lazy<Uri>(FormatHostAddress);
             _brokerAddress = new Lazy<Uri>(FormatBrokerAddress);
+            _connectionFactory = new Lazy<IConnectionFactory>(() => new NMSConnectionFactory(BrokerAddress));
         }
 
-        public string Host { get; set; }
-        public int Port { get; set; }
+        public Uri ConnectionString { get; set; }
         public string Username { get; set; }
         public string Password { get; set; }
-        public bool UseSsl { get; set; }
 
         public Uri HostAddress => _hostAddress.Value;
         public Uri BrokerAddress => _brokerAddress.Value;
 
         public IConnection CreateConnection()
         {
-            ITransport transport;

```

Comment:
This completely breaks AWS, which requires explicit TLS use.

Reply:
Here is the issue, MassTransit can't be used for any AmazonMQ config other than for the single instance, which is only suggested for development. I'm actually doing this because I need AmazonMQ support. The NMSConnectionFactory will actually select the appropriate transport for you based on your connection string, so you don't have to do what you were doing before and explicitly selecting a transport. If you get the connection string from AmazonMQ, will look something like this: failover:(ssl://host1:61617,ssl://host2:61617) and giving that to the connection factory will give you an SSL transport.

Reply:
This doc explains all the protocol options you can specify in the connection uri: http://activemq.apache.org/nms/activemq-uri-configuration.html

Reply:
The problem is, you can't force it to use Tls via the URI. If there is a way to do that, then this would work.

Reply:
I tried exactly to use ssl:// for amazon and it did not work.

Reply:
It does work. 

I just used this example online to test: http://activemq.apache.org/nms/nms-simple-asynchronous-consumer-example.html except the URI I used was: 

`activemq:failover:ssl://b-bcac9923-5808-44c0-9ee1-76c7b18bbe52-1.mq.us-west-2.amazonaws.com:61617`

I also tried:

`activemq:ssl://b-bcac9923-5808-44c0-9ee1-76c7b18bbe52-1.mq.us-west-2.amazonaws.com:61617`

They both connect and send and receive messages for me. That example uses the same NMSConnectionFactory I'm using right here.

Reply:
Out of curiosity, I tried the same connection using `tcp://` just to see and I got an exception so it looks like it's choosing the transport correctly.